diff --git a/flake.nix b/flake.nix
index af206a740e..f64b2c0eab 100644
--- a/flake.nix
+++ b/flake.nix
@@ -331,6 +331,7 @@
                   linuxPackages.cardano-wallet
                   linuxPackages.bech32
                   linuxPackages.cardano-address
+                  linuxPackages.local-cluster
                   cardano-node-1_35_4.hydraJobs.linux.musl.cardano-cli
                   cardano-node-1_35_4.hydraJobs.linux.musl.cardano-node
                 ];
@@ -339,6 +340,7 @@
                   "cardano-wallet"
                   "bech32"
                   "cardano-address"
+                  "local-cluster"
                   "cardano-cli"
                   "cardano-node"
                 ];
@@ -361,6 +363,7 @@
                       windowsPackages.cardano-wallet
                       windowsPackages.bech32
                       windowsPackages.cardano-address
+                      windowsPackages.local-cluster
                       cardano-node-1_35_4.hydraJobs.linux.windows.cardano-cli
                       cardano-node-1_35_4.hydraJobs.linux.windows.cardano-node
                     ];
@@ -387,6 +390,7 @@
                     macOsPkgs.cardano-wallet
                     macOsPkgs.bech32
                     macOsPkgs.cardano-address
+                    macOsPkgs.local-cluster
                     cardano-node-1_35_4.hydraJobs.macos.cardano-cli
                     cardano-node-1_35_4.hydraJobs.macos.cardano-node
                   ];
@@ -401,6 +405,7 @@
                     macOsPkgs.cardano-wallet
                     macOsPkgs.bech32
                     macOsPkgs.cardano-address
+                    macOsPkgs.local-cluster
                     cardano-node-1_35_4.hydraJobs.macos.cardano-cli
                     cardano-node-1_35_4.hydraJobs.macos.cardano-node
                   ];
