From 3f0186bdc9231b0bc3523294d4b6ead52b0e1419 Mon Sep 17 00:00:00 2001
From: Samuel Evans-Powell <mail@sevanspowell.net>
Date: Mon, 11 Jul 2022 10:14:31 +0800
Subject: [PATCH 1/2] Bump to cardano-node 1.35.1

---
 cabal.project | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cabal.project b/cabal.project
index 40f1ec8899..9eb3a27fd4 100644
--- a/cabal.project
+++ b/cabal.project
@@ -213,8 +213,8 @@ source-repository-package
 source-repository-package
     type: git
     location: https://github.com/input-output-hk/cardano-node
-    tag: 9f1d7dc163ee66410d912e48509d6a2300cfa68a
-    --sha256: 06arx9hv7dn3qxfy83f0b6018rxbsvh841nvfyg5w6qclm1hddj7
+    tag: c75451f0ffd7a60b5ad6c4263891e6c8acac105a
+    --sha256: 1z0zv1i58ikmbqg878f9z573jkwp4lzhmmswshm6c96rq6lprzh8
     subdir:
       cardano-api
       cardano-git-rev

From a4d31a3e3d4dbd3b713dff5d911edd3869a3b343 Mon Sep 17 00:00:00 2001
From: Samuel Evans-Powell <mail@sevanspowell.net>
Date: Mon, 11 Jul 2022 10:21:02 +0800
Subject: [PATCH 2/2] Bump version of cardano-node in compatibility matrix

---
 README.md     | 2 +-
 cabal.project | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/README.md b/README.md
index d10239eec3..b0c8c7ceec 100644
--- a/README.md
+++ b/README.md
@@ -73,7 +73,7 @@ See **Installation Instructions** for each available [release](https://github.co
 >
 > | cardano-wallet | cardano-node (compatible versions) | SMASH (compatible versions)
 > | --- | --- | ---
-> | `master` branch | [1.35.0](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.0) | [1.6.1](https://github.com/input-output-hk/smash/releases/tag/1.6.1)
+> | `master` branch | [1.35.1](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.1) | [1.6.1](https://github.com/input-output-hk/smash/releases/tag/1.6.1)
 > | [v2022-07-01](https://github.com/input-output-hk/cardano-wallet/releases/tag/v2022-07-01) | [1.35.0](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.0) | [1.6.1](https://github.com/input-output-hk/smash/releases/tag/1.6.1)
 > | [v2022-05-27](https://github.com/input-output-hk/cardano-wallet/releases/tag/v2022-05-27) | [1.34.1](https://github.com/input-output-hk/cardano-node/releases/tag/1.34.1) | [1.6.1](https://github.com/input-output-hk/smash/releases/tag/1.6.1)
 > | [v2022-04-27](https://github.com/input-output-hk/cardano-wallet/releases/tag/v2022-04-27) | [1.34.1](https://github.com/input-output-hk/cardano-node/releases/tag/1.34.1) | [1.6.1](https://github.com/input-output-hk/smash/releases/tag/1.6.1)
diff --git a/cabal.project b/cabal.project
index 9eb3a27fd4..c83a3e282c 100644
--- a/cabal.project
+++ b/cabal.project
@@ -183,8 +183,8 @@ source-repository-package
 source-repository-package
     type: git
     location: https://github.com/input-output-hk/cardano-ledger
-    tag: ce3057e0863304ccb3f79d78c77136219dc786c6
-    --sha256: 19ijcy1sl1iqa7diy5nsydnjsn3281kp75i2i42qv0fpn58238s9
+    tag: 3be8a19083fc13d9261b1640e27dd389b51bb08e
+    --sha256: 0dvm9l43mp1i34bcywmznd0660hhcfxwgawypk9q1hjkml1i41z3
     subdir:
       eras/alonzo/impl
       eras/alonzo/test-suite
