diff --git a/node_modules/usb/bin/darwin-x64-89/usb.node b/node_modules/usb/bin/darwin-x64-89/usb.node
new file mode 100755
index 0000000..8802ab9
Binary files /dev/null and b/node_modules/usb/bin/darwin-x64-89/usb.node differ
diff --git a/node_modules/usb/binding.gyp b/node_modules/usb/binding.gyp
index c926142..e2eba7d 100644
--- a/node_modules/usb/binding.gyp
+++ b/node_modules/usb/binding.gyp
@@ -10,7 +10,8 @@
       'cflags_cc!': [ '-fno-exceptions' ],
       'xcode_settings': { 'GCC_ENABLE_CPP_EXCEPTIONS': 'YES',
         'CLANG_CXX_LIBRARY': 'libc++',
-        'MACOSX_DEPLOYMENT_TARGET': '10.7',
+        'MACOSX_DEPLOYMENT_TARGET': '10.10',
+        'CLANG_CXX_LANGUAGE_STANDARD': 'c++14',
       },
       'msvs_settings': {
         'VCCLCompilerTool': { 'ExceptionHandling': 1 },
@@ -49,10 +50,10 @@
           }],
           ['OS=="mac"', {
             'xcode_settings': {
-              'OTHER_CFLAGS': [ '-std=c++1y', '-stdlib=libc++' ],
+              'OTHER_CFLAGS': [ '-std=c++14', '-stdlib=libc++' ],
               'OTHER_LDFLAGS': [ '-framework', 'CoreFoundation', '-framework', 'IOKit' ],
               'SDKROOT': 'macosx',
-              'MACOSX_DEPLOYMENT_TARGET': '10.7',
+              'MACOSX_DEPLOYMENT_TARGET': '10.10',
             },
           }],
           ['OS=="win"', {