diff --git a/flake.nix b/flake.nix
index e05d417f81..a985140f89 100644
--- a/flake.nix
+++ b/flake.nix
@@ -420,6 +420,7 @@
               inherit pkgs;
               project = walletProject;
             };
+            windowsPackages = mkPackages walletProject.projectCross.mingwW64;
           }) // {
             # Continuous integration builds
             ci.tests.all = pkgs.releaseTools.aggregate {
